src/chart/generateCategoricalChart.tsx

```diff
@@ -238,6 +239,21 @@ const getTooltipContent = (
       // graphic child has data props
       const entries = data === undefined ? displayedData : data;
       payload = findEntryInArray(entries, tooltipAxis.dataKey, activeLabel);
+    } else if (tooltipAxis.dataKey && tooltipAxis.allowSelectNearestValue && tooltipAxis?.categoricalDomain) {
+      const entries = data === undefined ? displayedData : data;
+      const value = tooltipAxis?.categoricalDomain?.[activeIndex];
+
+      const matchingData: any = entries.reduce((acc: any, entry: any) =>
+        Math.abs(value - entry[dataKey as string]) < Math.abs(value - acc[dataKey as string]) ? entry : acc,
```

Comment:
why does `dataKey` need to be cast?

Reply:
The dataKey can be an accessor function as well, right? Do we have an existing helper function to get the value from a row/entry using a dataKey?

Reply:
I believe the function you would prefer to use here, you can find in `src/util/ChartUtils.ts`

```
export function getValueByDataKey<T>(obj: T, dataKey: DataKey<any>, defaultValue?: any) {
  if (_.isNil(obj) || _.isNil(dataKey)) {
    return defaultValue;
  }

  if (isNumOrStr(dataKey)) {
    return _.get(obj, dataKey, defaultValue);
  }

  if (_.isFunction(dataKey)) {
    return dataKey(obj);
  }

  return defaultValue;
}
```